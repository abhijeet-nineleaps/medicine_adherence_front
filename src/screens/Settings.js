import {Linking, Share, View} from 'react-native';
import React from 'react';
import SettingsList from 'react-native-settings-list';
import styles from './screenStyles/SettingStyles';

const Settings = (navigation) => {
  const shareFnc = () => {
    try {
      Share.share({
        title: 'Medstick',
        message:
          'Hello you are invited to use Medstick ' +
          'https://play.google.com/store/apps/details?id=com.animesafar.dinterviewkit',
        url: 'https://cdn.discordapp.com/attachments/941592669933682699/955175698568462437/vinaylogo.png',
      });
    } catch (error) {}
  };
  const aboutFnc = () => {
    navigation.navigate('About');
  };
  const openSetFnc = () => {
    Linking.openSettings();
  };
  return (
    <View style={styles.container}>
      <SettingsList borderColor="white" backgroundColor="white">
        <SettingsList.Header
          headerText="Settings"
          headerStyle={styles.setting}
        />
        <SettingsList.Item
          hasNavArrow={true}
          title="Notification settings"
          titleStyle={styles.settingItems}
          id="openSetting"
          onPress={openSetFnc}
        />
        <SettingsList.Header
          headerText="General"
          headerStyle={styles.general}
        />
        <SettingsList.Item
          hasNavArrow={false}
          title="About Medstick"
          titleStyle={styles.settingItems}
          id="about"
          onPress={aboutFnc}
        />
        <SettingsList.Item
          hasNavArrow={false}
          title="Share with friends and family"
          titleStyle={styles.settingItems}
          id="share"
          onPress={shareFnc}
        />
      </SettingsList>
    </View>
  );
};
export default Settings;
