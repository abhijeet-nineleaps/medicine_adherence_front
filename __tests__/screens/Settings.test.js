import React from 'react';
import Enzyme from 'enzyme';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';
import Settings from '../../src/screens/Settings';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
Enzyme.configure({adapter: new Adapter()});

describe('Settings Screen', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Settings navigation={undefined} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('test share button', () => {
   const mockFn = jest.fn();
  const wrapper = shallow(<Settings shareFnc={mockFn} />);
  wrapper.find("#share").simulate("press");
  });
  it('test about button', () => {
    const mockFn = jest.fn();
   const wrapper = shallow(<Settings aboutFnc={mockFn} />);
   wrapper.find("#about").simulate("press");
   });
   it('test open setting button', () => {
    const mockFn = jest.fn();
   const wrapper = shallow(<Settings openSetFnc={mockFn} />);
   wrapper.find("#openSetting").simulate("press");
   });
});