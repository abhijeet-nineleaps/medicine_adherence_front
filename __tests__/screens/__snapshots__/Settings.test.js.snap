// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings Screen renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "height": "100%",
    }
  }
>
  <RCTScrollView>
    <View>
      <View>
        <Text
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "margin": 5,
              },
              Object {
                "color": "gray",
                "fontSize": 16,
              },
            ]
          }
        >
          Settings
        </Text>
        <View
          style={
            Object {
              "borderBottomWidth": 1,
              "borderColor": "white",
              "borderTopWidth": 1,
            }
          }
        >
          <View
            accessible={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
          >
            <View
              style={
                Array [
                  Object {
                    "flex": 1,
                    "flexDirection": "row",
                    "justifyContent": "center",
                  },
                  Object {
                    "backgroundColor": "white",
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "flexDirection": "row",
                      "marginLeft": 15,
                    },
                    Object {
                      "borderWidth": 0,
                    },
                    Object {
                      "minHeight": 50,
                    },
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "black",
                        "fontSize": 18,
                      },
                      Object {
                        "alignSelf": "center",
                        "flex": 1,
                      },
                    ]
                  }
                >
                  Notification settings
                </Text>
                <Image
                  source={
                    Object {
                      "testUri": "../../react-native-settings-list/img/icon-arrow-settings.png",
                    }
                  }
                  style={
                    Array [
                      Object {
                        "alignSelf": "center",
                        "marginRight": 15,
                      },
                      undefined,
                    ]
                  }
                />
              </View>
            </View>
          </View>
        </View>
      </View>
      <View>
        <Text
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "margin": 5,
              },
              Object {
                "color": "gray",
                "fontSize": 16,
              },
            ]
          }
        >
          General
        </Text>
        <View
          style={
            Object {
              "borderBottomWidth": 1,
              "borderColor": "white",
              "borderTopWidth": 1,
            }
          }
        >
          <View
            accessible={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
          >
            <View
              style={
                Array [
                  Object {
                    "flex": 1,
                    "flexDirection": "row",
                    "justifyContent": "center",
                  },
                  Object {
                    "backgroundColor": "white",
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "flexDirection": "row",
                      "marginLeft": 15,
                    },
                    Object {
                      "borderBottomWidth": 1,
                      "borderColor": "white",
                    },
                    Object {
                      "minHeight": 50,
                    },
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "black",
                        "fontSize": 18,
                      },
                      Object {
                        "alignSelf": "center",
                        "flex": 1,
                      },
                    ]
                  }
                >
                  About Medstick
                </Text>
              </View>
            </View>
          </View>
          <View
            accessible={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
          >
            <View
              style={
                Array [
                  Object {
                    "flex": 1,
                    "flexDirection": "row",
                    "justifyContent": "center",
                  },
                  Object {
                    "backgroundColor": "white",
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "flexDirection": "row",
                      "marginLeft": 15,
                    },
                    Object {
                      "borderWidth": 0,
                    },
                    Object {
                      "minHeight": 50,
                    },
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "black",
                        "fontSize": 18,
                      },
                      Object {
                        "alignSelf": "center",
                        "flex": 1,
                      },
                    ]
                  }
                >
                  Share with friends and family
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
